sirepo/job_supervisor.py

```diff
@@ -848,13 +858,9 @@ async def _send_with_single_reply(self, opName, req, **kwargs):
                     # POSIT: any api_* that could run into runDirNotFound
                     # will call _send_with_single_reply() and this will
                     # properly format the reply
-                    if not r.get('runDirNotFound'):
-                        return r
-                    self.__db_init(req, prev_db=self.db)
-                    self.__db_write()
-                    assert self.db.status == job.MISSING, \
-                        'expecting missing status={}'.format(self.db.status)
-                    return PKDict(state=self.db.status)
+                    if r.get('runDirNotFound'):
```

Comment:
what if runDir is there, but no out.json.

Reply:
Then the reading of the results will fail and the the `r.state == job.ERROR` check in runStatus will catch the error. 

Thinking about this more I think it would be best to eliminate runDirNotFound and move the `r.state == job.ERROR` check and handling that I added to runStatus into here. All callers of send_with_single_reply I think would benefit from this logic. I think every caller of send_with_single_reply will need the compute to be re-run if they get back an error from the agent. 

My assumption in all of this is that our code "works" and that when we get back error it really is an error with job. Not some transient error in our system (ex fastcgi socket dropped or we didn't send properly formatted json). Is this a safe assumption to make? If not should we create two classes of error (1 for error in our system and 1 for error with the actual compute/results)?

Reply:
I don't think we need to assume one error or the other or have two classes of error. I do think we should interpret the errors in the GUI (or perhaps server-side) to be make more sense to the end user. We obviously can't do this always. 

I think we should also try to prevent errors percolating so if we know an out.json doesn't exist, we can just translate that to "missing", since it is probably not an end user serviceable error. We just try to do our best.

Reply:
I think runDirNotFound is fine to do this way.